python_language/filter_words/filter_words.py

```diff
@@ -0,0 +1,38 @@
+"""
+Filter Words
+------------
+
+Print out only words that start with "o", ignoring case::
+
+    lyrics = '''My Bonnie lies over the ocean.
+                My Bonnie lies over the sea.
+                My Bonnie lies over the ocean.
+                Oh bring back my Bonnie to me.
+                '''
+
+Bonus points: print out words only once.
+
+See :ref:`filter-words-solution`.
+"""
+import re
+lyrics = '''My Bonnie lies over the ocean.
+            My Bonnie lies over the sea.
+            My Bonnie lies over the ocean.
+            Oh bring back my Bonnie to me.
+         '''
+lyrics=lyrics.lower()
+lyrics=re.sub(r'[\.,]?','',lyrics)
+lyricList=lyrics.split()
+owords=[]
+for word in lyricList:
+        if(word[:1]=='o'):
```

Comment:
There is a `str` method `str.startswith()` that is the "Pythonic" way of checking if a string starts with another string.  Usually, but not always, the method of a data type is faster and more efficient than using an indexing based algorithm.  Here, the check would be `if word.startswith('o'):`